src/LearnMe.Core/Services/Calendar/GoogleCalendar.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using LearnMe.Core.DTO.Calendar;
+using LearnMe.Core.Interfaces.Services;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Google.Apis.Calendar.v3;
+using Google.Apis.Calendar.v3.Data;
+using LearnMe.Core.Services.Calendar.Utils;
+using Microsoft.Extensions.Logging;
+using LearnMe.Infrastructure.Models.Domains.Calendar;
+using LearnMe.Infrastructure.Repository.Interfaces;
+
+namespace LearnMe.Core.Services.Calendar
+{
+    public class GoogleCalendar : ICalendar
+    {
+        private readonly CalendarService _calendarService;
+        private readonly string _applicationName = "Learn Me WEB Applicaton";
+        private readonly ICrudRepository<CalendarEvent> _repository;
+        private readonly IMapper _mapper;
+        private readonly ILogger<GoogleCalendar> _logger;
+
+        public GoogleCalendar(IGoogleAPIconnection googleAPIconnection,
+                              ICrudRepository<CalendarEvent> repository,
+                              IMapper mapper,
+                              ILogger<GoogleCalendar> logger)
+        {
+            var token = googleAPIconnection.GetToken();
+            _calendarService = googleAPIconnection.CreateCalendarService(token, _applicationName);
```

Comment:
Polecam zapoznać się ze wzorcem projektowym `Unit Of Work`. Ten kod nie jest zły, tylko można go zrobić ładniej ;P Wtedy warto zmienić nazwę zmiennej i interfejsu na `_calendarUow`. W skrócie, Unit Of Work to zbiór serwisów w jednej klasie (serwisy, repozytoria itp)

https://www.programmingwithwolfgang.com/repository-and-unit-of-work-pattern/